pkg/client/client.go

```diff
@@ -864,6 +889,59 @@ func (c *Client) WaitForPrometheus(ctx context.Context, p *monv1.Prometheus) err
 	return nil
 }
 
+func (c *Client) ValidatePrometheus(ctx context.Context, promNsName types.NamespacedName) error {
+	var validationError error
+
+	pollErr := wait.Poll(10*time.Second, 5*time.Minute, func() (bool, error) {
+
+		p, err := c.GetPrometheusByNsName(ctx, promNsName)
+		if err != nil {
+			klog.V(4).Info("validate prometheus failed to get prometheus: ", err)
+			return false, err
+		}
+
+		avail, err := prometheusConditonForType(p.Status.Conditions, monv1.PrometheusAvailable)
+		if err != nil {
+			validationError = NewUnavailableError(err.Error())
+			return false, nil
+		}
+
+		if avail.Status == monv1.PrometheusConditionTrue {
+			return true, nil // terminate if available
+		}
+
+		// return reason for failure as state-errors - Degraded/Unavailable
+		errs := []error{}
+		msg := fmt.Sprintf("%s: %s", avail.Reason, avail.Message)
+
+		newStateError := NewDegradedError
+		if avail.Status == monv1.PrometheusConditionFalse {
+			// prometheus not available is a Degraded and Unavailable error
+			errs = append(errs, NewDegradedError(msg))
+			newStateError = NewUnavailableError
```

Comment:
Not sure I follow this. If the available condition is false (or unavailable), to that is a stronger statement then degraded. Degraded tells me that it _is_ available, just not at the level specified.

Reply:
hm I think I misunderstood. In p-o the `Available` condition can be `Degraded`.
I'm not sure we want to set the `Degraded` condition here. Reconciliation might have worked fine and Prometheus might be unavailable for other reasons.

Reply:
This actually aligns with our earlier discussions that Prometheus becoming unavailable should set `Degraded` condition of CMO to `true`.

Reply:
@jan--f Marking this as resolved. Please reopen if you want to discuss the comment above further. 

> This actually aligns with our earlier discussions that Prometheus becoming unavailable should set Degraded condition of CMO to true.

Reply:
Added - reconciled condition check as well.